### ISSUE TITLE
QAMP Project Proposal: Impact of Noise on QML algorithms

### ISSUE THREAD

**Issue by muf148** [qiskit-advocate/qamp-2025#7]
---
name: 'Muhammad Faryad'
about: 'Submit a project proposal for QAMP '
title: 'Impact of Noise on QML algorithms'
labels: project-proposal
assignees: ''

---

# QAMP Project Proposal: Impact of Noise on QML algorithms

## Description
The project will focus on understanding the role of noise in the expressibility and complexity of QML algorithms. On the theoretical side, we will develop theorems to construct quantum models in the presence of simple noise models. Simulation work will also be required to compare the noisy and noiseless implementations of the QML models. The mentees are supposed to be well-versed in the theory of QML algorithms, density matrices, and quantum channels representation of noise models.

## Deliverables
**Primary Deliverables:**
- A latex report in PRA format containing Intro, Proofs of Theorem, Numerical Results, and References. 
- A Jupyter Notebook containing the code.  

## Mentor

**Name:** Muhammad Faryad
**GitHub:** @muf148
**What I do:** I am an associate professor of physics at LUMS University in Pakistan. I teach courses on QC and QI and supervise research work of BS, MS, and PhD students.

**Mentoring approach:** I provide detailed guidelines, help with calculations, look at the codes, discuss the results, and contribute to writing.

**Type of mentees sought:** I am looking for a mentee with a background in quantum information with knowledge of density matrix formalism and noise channels. An experience with Qiskit and QML algorithms is also required.

**Additional Information:**
- **Time commitment:** 10-20 hours a week, depending upon the background.

**Comment by muf148** [comment]
Thanks, @Tundekushimo 

I would love to work with you. Please apply when the mentee applications open soon. I have one other mentee for this project and I will work with both of you.

**Comment by Hasnainfs** [comment]
**Application for "Impact of Noise on QML Algorithms" Project**

Dear Sir,

My name is Muhammad Hasnain. I completed my MPhil in Electronics from Quaid-i-Azam University in 2021 under the supervision of Dr. Rizvi. Currently, I am working with Qiskit and have a strong interest in quantum noise and error mitigation.

I can commit 10–15 hours per week to this project and am very eager to work under your supervision. Please kindly consider me for this opportunity.

Sincerely,
Muhammad Hasnain

**Comment by muf148** [comment]
Anyone who is interested, please submit your mentee application, opening tomorrow.

**Comment by quantum-kittens** [comment]
Great to see all the interest! Prospective mentees: please do not forget to also submit the mentee application form and select this project in your application. [Here](https://github.com/qiskit-advocate/qamp-2025/issues/30) is guidance on becoming a mentee.

**Comment by Mr-Jamatia** [comment]
Keen to be part of it

**Comment by JeromeB7** [comment]
Hello Dr. @mufi48,

I am applying to the QAMP 2025 program as a mentee and wanted to express my strong interest in this project as one of my top choices.

During the Qiskit Global Summer School, I conducted a deep study on Quantum Noise and Error Mitigation, which gave me a strong theoretical foundation for the challenges in the NISQ era. This project is the perfect opportunity for me to apply that theoretical knowledge to a practical, hands-on research problem.

This looks like a fantastic and highly relevant study. Looking forward to the possibility of contributing!

**Comment by muf148** [comment]
Dear Mentees,

Please apply. I will try to supervise upto 3 people.

Thanks a lot for your interests.

Faryad

**Comment by Thirumalai-97** [comment]
Dear Muhammad Faryad, 

I’m Thirumalai. I come from a multidisciplinary background spanning Quantum Technologies, Computer Science, and Electronics. My academic and research work has primarily focused on designing tunable parametric quantum circuits, quantum optimization, and quantum machine learning.

Please visit my site for more details - https://thirumalai-97.github.io/portfolio/

I would like to work under this project. Thanks for consideration.

**Comment by bhavnaboseQuantum** [comment]
Impact of Noise on Quantum Machine Learning Algorithms QAMP Issue # 7 Check point 2
Bhavna Bose
QAMP Mentee

Dr. Muhammad Faryad
QAMP Mentor
Abstract—Quantum Machine Learning (QML) combines quantum computing principles with machine learning models to potentially achieve computational advantages. However, current quantum hardware is highly susceptible to noise arising from decoherence, gate errors, and environmental interactions. This work investigates the impact of different quantum noise models on variational quantum machine learning algorithms for binary classification. We analyze how circuit depth, qubit count, and feature encoding strategies influence robustness under noisy conditions. Experimental results using Qiskit-based simulations demonstrate that noise significantly degrades performance, emphasizing the need for noise-aware circuit design in the NISQ era.
Index Terms—Quantum Machine Learning, Noise Models, Variational Quantum Circuits, Quantum Neural Networks, NISQ
I. PROJECT OVERVIEW
The goal of this project is to investigate how noise affects the performance of Quantum Machine Learning (QML) algorithms, particularly for binary classification tasks. While QML models show promising results in ideal, noiseless simulators, real quantum hardware is highly susceptible to noise arising from decoherence, gate errors, and environmental interactions. As circuit depth and qubit count increase, noise accumulation significantly degrades performance. Our project focuses on systematically analyzing how different quantum noise models—such as depolarizing noise, amplitude damping noise, and phase damping noise—impact variational quantum circuits (VQCs) and quantum neural networks(QNNs). We also study how circuit design choices, including feature maps and number of qubits, influence robustness under noisy conditions. The long-term objective is to derive noise aware insights that can guide practical QML circuit design in the NISQ era.
II. PROGRESS MADE
1) Significant progress has been achieved since Checkpoint 1:
Noise Modeling on Quantum Circuits We successfully implemented multiple noise models using Qiskit Aer, including depolarizing noise, amplitude damping noise,and phase damping noise. These noise channels wereapplied to quantum circuits to observe their individualand combined effects.
2) Single-Qubit Variational Circuit Experiments A simple1-qubit variational quantum circuit was implemented. We recorded classification outputs and expectation values both before and after adding noise. Clear deviations were observed when noise was introduced, confirming the sensitivity of even shallow circuits.
3) Multi-Qubit QML Circuit Implementation We extended the experiments to a multi-qubit setting using a 3-qubit circuit. Classical features were encoded using a ZZ Feature Map, followed by a variational ansatz. The impact of noise on model accuracy and convergence behavior was recorded.
4) Binary Classification Pipeline A working binary classification pipeline using a quantum neural network was implemented and evaluated under different noise configurations.
III. EXPLORATIONS AND EXPERIMENTS
During development, several approaches were explored: Initially, we attempted to use high-level abstractions from the Qiskit Machine Learning library (VQC, QSVM). However, version incompatibilities between Qiskit releases caused repeated failures and inconsistent behavior.
Adding noise directly to transpiled circuits proved challenging due to backend constraints and API changes. As an alternative, we implemented a lower-level Quantum Neural Network approach, which provided finer control over circuit construction, noise injection, and training loops. This approach was successful and became the primary experimental path.
What worked well was building modular circuits and gradually increasing complexity. What did not work reliably was mixing Qiskit versions or relying heavily on deprecated APIs.
IV. KEY INSIGHTS AND LEARNINGS
Several important insights emerged from this work:
QML models are expressive but extremely sensitive to noise, even for shallow circuits. Increasing the number of qubits and entangling gates improves representational power but significantly amplifies noise effects.
Phase damping noise had a particularly strong impact on circuits relying on interference patterns. Achieving quantum advantage requires careful fine-tuning; na¨ıvely scaling circuits often leads to worse performance than classical baselines.
Noise-aware circuit design is essential for realistic QML experimentation. These findings reinforced the idea that robustness, not just expressivity, is critical in near-term quantum machine learning.
V. TECHNICAL CHALLENGES
The most significant technical challenge was managing Qiskit version compatibility. Mixing pre-1.0 and post-1.0 Qiskit packages led to import errors and unexpected behavior. This was addressed by creating clean virtual environments and carefully pinning compatible versions. Another challenge was injecting noise into transpiled circuits. This required deeper understanding of Qiskit Aer noise models and backend execution flow. Moving to a custom QNN implementation helped overcome this limitation.
VI. CURRENT STATUS
At present, the project has:
1) A working 3-qubit quantum neural network for binary classification
2) ZZ Feature Map–based data encoding
3) Multiple noise models successfully integrated
4) Comparative results for noiseless vs noisy circuits
5) The experimental pipeline is functional and reproducible.
VII. PATH FORWARD (TOWARD MVP BY JANUARY 30)
To complete the MVP, the following steps are planned:
1) Design a finalized QML circuit with explicit transpilation and noise modeling
2) Systematically vary noise strengths and analyze performance trends
3) Add quantitative evaluation metrics (accuracy, loss curves)
4) Finalize a clean, well-documented Jupyter notebook
5) Prepare a short demo showcasing noisy vs noiseless results
6) Begin drafting a research paper based on the findings
This will result in a complete, noise-aware QML benchmarking MVP.
